platform-sdk/swirlds-common/src/main/java/com/swirlds/common/context/PlatformContext.java

```diff
@@ -63,4 +67,44 @@ public interface PlatformContext {
      */
     @NonNull
     Time getTime();
+
+    /**
+     * Returns the {@link ExecutorFactory} instance for the platform
+     *
+     * @return the {@link ExecutorFactory} instance
+     */
+    ExecutorFactory getExecutorFactory();
+
+    /**
+     * Creates a new instance of the platform context. The instance uses a {@link NoOpMetrics} implementation for
+     * metrics. The instance uses the static {@link CryptographyHolder#get()} call to get the cryptography. The instance
+     * uses the static {@link Time#getCurrent()} call to get the time.
+     *
+     * @param configuration the configuration
+     * @return the platform context
+     * @deprecated since we need to remove the static {@link CryptographyHolder#get()} call in future.
+     */
+    @Deprecated(forRemoval = true)
+    @NonNull
+    static PlatformContext create(@NonNull final Configuration configuration) {
+        return DefaultPlatformContext.create(configuration);
+    }
+
+    /**
+     * Creates a new instance of the platform context.
+     * <p>
+     * The instance uses the static {@link Time#getCurrent()} call to get the time.
+     *
+     * @param configuration the configuration
+     * @param metrics       the metrics
+     * @param cryptography  the cryptography
+     * @return the platform context
+     */
+    @NonNull
+    static PlatformContext create(
+            @NonNull final Configuration configuration,
+            @NonNull final Metrics metrics,
+            @NonNull final Cryptography cryptography) {
+        return DefaultPlatformContext.create(configuration, metrics, cryptography);
+    }
```

Comment:
Is there a reason why we don't allow all of the utilities in the context to be provided when we create a new context?

Reply:
The only reason is that it is not used today :D
I assume for turtle we want to have that but for now I only created what is really needed

Reply:
Makes sense. We can add capability as we need it.